.github/workflows/link-check-deployment-status.yml

```diff
@@ -0,0 +1,56 @@
+name: Check new links against deployment
+on:
+  workflow_call:
+    inputs:
+      main:
```

Comment:
- could rename this to `branch`?
- or better, auto-detect via something like `gh repo view -R ${{ github.event.deployment.payload.repository.full_name }} --json defaultBranchRef --jq .defaultBranchRef.name` (untested)

Reply:
I'll have to poke around in the data structure again, there's got to be some way we can find the main branch even if it takes an API call or two, but maybe even without one.

Reply:
Found it in an old context dump, `github.event.repository.default_branch` is what we want.

Reply:
https://github.com/iterative/link-check/pull/18 and https://github.com/iterative/cml.dev/pull/287 are now opened for this!